Cron/Feed.php

```diff
@@ -303,15 +311,27 @@ public function execute()
                         }
 
                         $this->printLine($siteId, '<mpn><![CDATA['.$this->escapeString($product->getSku()).']]></mpn>', 5);
-
-                        $stock_item = $this->_stockItem->get($product->getId());
-                        $this->printLine($siteId, '<stock>'.($stock_item && $stock_item->getIsInStock() ? ($stock_item->getMinQty() > 0 ? (int) $stock_item->getQty() : 1) : 0).'</stock>', 5);
+                        $stock_item = null;
+
+                        try {
+                            $stock_item = $this->_stockItem->get($product->getId());
+                            $this->printLine($siteId, '<stock>'.($stock_item && $stock_item->getIsInStock() ? ($stock_item->getQty() > 0 ? (int) $stock_item->getQty() : 1) : 0).'</stock>', 5);
+                        } catch (\Magento\Framework\Exception\NoSuchEntityException $e) {
+                            $stock_qty = $stock_state->getStockQty($product->getId());
+                            $this->printLine($siteId, '<stock>' . ($stock_qty ? $stock_qty : 0) .'</stock>', 5);
+                        }
 
                         $this->printLine($siteId, '<link>' . $product->getProductUrl(true) . '</link>', 5);
 
                         $image = $product->getImage();
-                        if (! empty($image)) {
-                            $this->printLine($siteId, '<image>' . $mediaUrl . 'catalog/product' . $image . '</image>', 5);
+                        if (empty($image) || $image == 'no_selection') {
+                            $image = $product->getMediaGalleryImages()->getFirstItem()->getUrl();
```

Comment:
Does this need to be done on the other bit as well? Also is it worth putting in a separate function like magento 1?

Reply:
Yeah, I've added a separate function.